### ISSUE TITLE
Objects off page problem

### ISSUE THREAD

**Issue by cleblond** [partridgejiang/Kekule.js#145]
If I create some objects in composer in a composer on a big display and then I load those objects into a composer on a smaller display some of the objects are off the page and cannot be scrolled to.  Any way to center and zoom in automatically on loading in such cases?

![kjs_object_off_page](https://user-images.githubusercontent.com/4019904/68242064-d2bd7e00-ffdd-11e9-8153-7ca9d9d8635e.png)

**Comment by partridgejiang** [comment]
Sorry that I have been "isolated" from Internet for several weeks for involving in a state examination. Now the composer widget will automatically enlarge the client to display the large objects. Of course this feature can also be turned off by code:
```js
composer.getEditorConfigs().getInteractionConfigs().setAutoExpandSizeAfterLoading(false);
```
Please check the latest dist files.

btw,  the bond forming electron pushing arrow is also in implementation now.

**Comment by partridgejiang** [comment]
A new feature has been implemented in the composer. The client size will also be expanded if objects are dragging outside the edge of the editor. The feature is off by default but you can manually turn it on with:
```js
composer.getEditorConfigs().getInteractionConfigs().setAutoExpandClientSizeAfterManipulation(true);
```

**Comment by cleblond** [comment]
Thanks much I'll test this week!  Hope your exams went well!

**Comment by yitzchak** [comment]
The resizing of the canvas definitely helps. 

Additionally, automatic setting of the zoom and pan would be useful so that the whole structure can be seen on load. In addition, having a button that "fits" the viewport to the structure would also be really useful. Is there any functionality for that?

**Comment by partridgejiang** [comment]
@yitzchak, thanks for the excellent suggestion. It should be  implemented soon, :).

**Comment by agrodet** [comment]
Hello. Has the suggestion above been implemented? Or is there a way to scroll / zoom to the molecule position manually?

**Comment by partridgejiang** [comment]
@agrodet, yes, method ```autoExpandChemSpaceSize``` is introduced to expand the page size for containing all structures:
```js
composer.getEditor().autoExpandChemSpaceSize();
```
The ```scrollClientToObject``` method can be used to manually scroll to specified molecule(s)/object(s) in editor:
```js
composer.getEditor().scrollClientToObject(molecule);
composer.getEditor().scrollClientToObject([molecule1, molecule2, object3]);
```

**Comment by agrodet** [comment]
Thank you for your prompt answer. It seems that I cannot used those methods. I'm using the following code, but nothing happens when I call any of the methods you suggested. Am I doing something wrong?
```
const composer = new Kekule.Editor.Composer(document); 
composer.setDimension('600px', '400px');
composer.setPredefinedSetting('molOnly');
const mol = Kekule.IO.loadFormatData(
    baseCompound,
    'mol'
);
this.composer.setChemObj(mol);
```

`composer.getEditorConfigs().getInteractionConfigs().setEditorInitialZoom()` is currently enough to meet my needs, but I'm curious as to why I cannot use methods that looks very much appropriate.

**Comment by partridgejiang** [comment]
@agrodet, please check the Kekule.VERSION value in your environment. The latest version is 0.9.8. If your release is lagged behind, please update the it with files in ```dist``` directory of this repo, or with npm (currently the published release on npm is 0.9.7, not the latest, but it still shipped with those methods).
